org.eclipse.xtext.ui/src/org/eclipse/xtext/ui/editor/findrefs/ReferenceSearchResultContentProvider.java

```diff
@@ -197,10 +199,28 @@ public IStatus runInUIThread(IProgressMonitor monitor) {
 				}
 			}
 			viewer.refresh();
-			viewer.expandToLevel(1);
+			expandToFirstChild();
 			return Status.OK_STATUS;
 		}
 	}
+	
+
+	private void expandToFirstChild() {
+		if (rootNodes != null && !rootNodes.isEmpty()){
```

Comment:
personally I prefer fast returns instead of an if-block that spans the whole method body